src/WebJobs.Extensions.Http/HttpDirectRequestBindingProvider.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using Newtonsoft.Json;
+using static Microsoft.Azure.WebJobs.Extensions.Http.HttpTriggerAttributeBindingProvider.HttpTriggerBinding;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    // support allowing binding any non-attributed HttpRequestMessage parameter to the incoming http request.     
+    //    Foo([HttpTrigger] Poco x, HttpRequestMessage y); 
+    internal class HttpDirectRequestBindingProvider : IBindingProvider
+    {
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            ParameterInfo parameter = context.Parameter;
+            if (parameter.ParameterType == typeof(HttpRequestMessage))
+            {
+                // Not already claimed by another trigger?
+                if (parameter.GetCustomAttributes(false).Length == 0)
```

Comment:
Should we be more specific on the type of attribute here (verify this is indeed a binding attribute so we still allow for other attributes)?